### DISCUSSION TITLE
Option to clear queue when playing new media

### DISCUSSION THREAD

**Discussion by agdamsbo** [music-assistant/support#1281] [Feature requests and ideas]
Hi! Would it be possible to have an option to set default behaviour to clear queue and start playing new item when choosing "play"? Current behavior is that the new item is starting to play, but the old queue is saved at the end. 

It would help when starting a new track or album with a script and having the player set to repeat queue.

**Comment by OzGav** [comment]
Makes sense. Let me see what the others think...

**Comment by userMak** [comment]
I think it would be very useful feature. Most of the times I add a playlist in repeat mode. (To start from the begging when it ends). However when the playlist ends MA continues to play the songs from a previous playlist which is wrong. If possible I would like at least an option if the user make the selection to clear the queue for example every time the media player is turning off or on. Something like that. One more thing is that the player in mobile version doesn't have the the option to clear the queue even manual. (at least I can not find it.)It would be useful to add it.

**Comment by userMak** [comment]
Can we have some feedback if this is planned to implement sometime?

**Comment by OzGav** [comment]
There was agreement internally to improve the queue functionality however right now the project is in bug squash mode not functionality enhance,ent

**Comment by marcelveldt** [comment]
the default behavior should already be to overwrite the existing queue if you start a new playlist or album. Only if you play a single track or radio station it will be inserted at the top of the existing queue (and played). Where is this playback started from ?

**Reply by userMak** [reply]
The playback starts correctly. But everything remains in the queue. One of the problems with this function is when someone wants to replay the selected playlist when it ends not to be possible, because after the selected playlist MA continues to play the rest songs or radio stations or whatever remained in the queue from previous days. I don't know how easy or difficult it is to implement and of course I'm in no rush.
I am mainly asking IF this going to implement sometime, because @OzGav  said above "Let me see what the others think...". That's why I am asking if you have decide about it.. I think there is no option that this could be achieved by an automation right?

**Comment by OzGav** [comment]
> the default behavior should already be to overwrite the existing queue if you start a new playlist or album.

I am editing this as I just relooked at the frontend. Looks like @marcelveldt has added to the options. @userMak you should find there is a `Play Now (clear queue) `option which is what you want?

**Reply by userMak** [reply]
I am not at home right now. Where is this option and at which version are you referring? (I am at 0.39 beta if I remember correct).
I tried to look through the companion app but it is even more difficult to operate MA. Most options are not available. (eg I can not manually clear the queue)

**Reply by OzGav** [reply]
Multiple places. The second one is using the 3dot menu on the right

![image](https://github.com/music-assistant/hass-music-assistant/assets/19848947/222ab47f-673c-4016-a26c-d5ea9124d486)

![image](https://github.com/music-assistant/hass-music-assistant/assets/19848947/791c9089-a3e7-4433-bc8f-84ecccdf7c28)

**Reply by userMak** [reply]
ok. I haven't see that. It nice addition! I would prefer to have an option in the settings to clear the queue for example each time a new playlist or radio station is added in order to be able to use it with phone also. 

Lately I am using the media player of my mobile device to start MA. I will check if version 0.39 has this option in my pc otherwise I will update it. This is what I get from companion app. Just play.. No option to clear the queue.

![image](https://github.com/music-assistant/hass-music-assistant/assets/55662933/754291b9-8f52-47a1-840b-f6068553c800)

Thanks!

**Reply by OzGav** [reply]
Yeah we are up to 51 now so a lot of changes

**Reply by userMak** [reply]
Sometime later or much later, please see the mobile version also.. It sure is behind the desktop version
Thanks again!

**Reply by OzGav** [reply]
I just tried the Android app and you can long press on the item to get the context menu and the show info screen is the same as on the desktop

**Reply by userMak** [reply]
I have Galaxy A52s and the long press doesn't work.. I don't know if it phone related or what but it doesn't work for me

**Reply by OzGav** [reply]
Hmmm... Also when you click that Play button in your first image that is when the four option menu should appear. If it doesn't then definitely update to b51 as frontend changes occur in these updates as well.

**Reply by userMak** [reply]
ok, I will update later, when I return home and I will let you know.
I am sure though that until now this option (long press) never worked for me in the past. I hope .51 solves it

**Reply by userMak** [reply]
Although I usually update when at home, I was curious about it, so I just updated HA from android app. I also restarted HA etc. The long press button still doesn't work for me. Until now I thought that it was problem with MA mobile version. Now that I learn that it is working for you I am not sure how I can solve it..

**Reply by OzGav** [reply]
Yeah. I tried it with my Android Phone (Pixel 4a) and my iPad and both worked. Have you got the latest version of the companion app as well? I just tried with Chrome on my phone and that also worked?

**Reply by OzGav** [reply]
Maybe its a phone option? Has Samsung done one their tweaks to the UI?

**Reply by userMak** [reply]
Yes, they use One UI (its theirs). I am not sure if I can make changes :(

**Reply by OzGav** [reply]
TBH this actually should just be an app function where it implements a long click listener so I am not sure why it isn't working for you. If you have another device to try...?

**Reply by userMak** [reply]
It doesn't work on my phone at all. I also tried the sumsung and edge browser. I will try me wife phone (redmi) when I return. I don't think I can do something though in my samsung.

**Reply by OzGav** [reply]
This works by selecting PLAY NOW from the play menu in beta60

**Reply by userMak** [reply]
Good morning. Yes now is working from my phone also. Thanks!

I noticed that the app clears the queue when selecting from a playlist, but not when choosing a radio station. It would be better to have consistent behavior for both actions, though it's not a major issue. Thanks again

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
